test/JDBC/expected/BABEL-3865.out

```diff
@@ -0,0 +1,140 @@
+/* Test a variety of normal/pg datatypes */
+CREATE TABLE #temp_1(
+	a int,
+	b date,
+	c varchar,
+	d text,
+	e json,
+	f xml
+)
+GO
+
+/* Test a variety of babelfish-specific datatypes */
+CREATE TABLE #temp_2(
+	a sysname,
+	b datetime,
+	c datetime2,
+	d nchar,
+	e nvarchar,
+	f varbinary
+)
+GO
+
+/* Test user-created datatypes */
+CREATE TYPE typa FROM int
+GO
+
+CREATE TYPE typb FROM nvarchar(100)
+GO
+
+CREATE TABLE #temp_3(
+	a typa,
+	b typb
+)
+GO
+
+DROP TYPE typb
+GO
+~~ERROR (Code: 3732)~~
+
+~~ERROR (Message: cannot drop type typb because other objects depend on it)~~
+
+
+DROP TYPE typa
+GO
+~~ERROR (Code: 3732)~~
+
+~~ERROR (Message: cannot drop type typa because other objects depend on it)~~
+
+
+DROP TABLE #temp_3
+DROP TYPE typb
+GO
+
+DROP TYPE typa
+GO
+
+/* Test computed columns */
+CREATE TABLE #temp_3_5(
+	a int,
+	b as a + 1)
+GO
+
+INSERT INTO #temp_3_5(a) VALUES (1)
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * FROM #temp_3_5
+GO
+~~START~~
+int#!#int
+1#!#2
+~~END~~
+
+
+DROP TABLE #temp_3_5
+GO
+
+
+/* Test SELECT INTO */
+CREATE TYPE typc FROM nchar
+GO
+
+CREATE TABLE tab_1 (a int, b typc)
+GO
+
+INSERT INTO tab_1 VALUES (1, 'a')
+GO
+~~ROW COUNT: 1~~
+
+
+SELECT * INTO #temp_4 FROM tab_1
+GO
+
+SELECT * FROM #temp_4
+GO
+~~START~~
+int#!#nchar
+1#!#a
+~~END~~
+
+
+DROP TYPE typc
+GO
+~~ERROR (Code: 3732)~~
+
+~~ERROR (Message: cannot drop type typc because other objects depend on it)~~
+
+
+DROP TABLE tab_1
+GO
+
+DROP TABLE #temp_4
+GO
+
+DROP TYPE typc
+GO
+
+/* Test Ownership + Roles, for pg_shdepend */
+CREATE ROLE role_a
+GO
+
+CREATE TABLE #temp_5 (a int)
```

Comment:
@timchang514

Reply:
yes, it goes into ENR. so there is one peculiar case where a role could be dropped from a different session but not from the local session with temp tables. I think this is ok to have, since functionality isn't lost if the role is dropped from a separate session. I also don't see a simple way to resolve this